packages/wrangler/src/dev.tsx

```diff
@@ -600,24 +603,22 @@ export async function startDev(args: StartDevOptions) {
 			);
 		}
 
-		let experimentalAssetsOptions = processExperimentalAssetsArg(args, config);
-		if (experimentalAssetsOptions) {
-			args.forceLocal = true;
-		}
-
-		/*
-		 * - `config.legacy_assets` conflates `legacy_assets` and `assets`
-		 * - `args.legacyAssets` conflates `legacy-assets` and `assets`
-		 */
 		if (
 			(args.legacyAssets || config.legacy_assets) &&
-			experimentalAssetsOptions
+			(args.site || config.site)
 		) {
 			throw new UserError(
-				"Cannot use Legacy Assets and Experimental Assets in the same Worker."
+				"Cannot use Legacy Assets and Workers Sites in the same Worker."
 			);
 		}
 
+		verifyMutuallyExclusiveAssetsArgsOrConfig(args, config);
```

Comment:
we should always perform this validation **before** we validate that the assets dir exists. IMHO this ordering makes more sense, since we'd want to first establish which of the mutually exclusive config keys/args the user actually meant, and once we're clear on that, we can proceed to perform validation specific to that config/arg